File: library/src/main/java/me/proxer/library/entity/notifications/NotificationInfo.java
package me.proxer.library.entity.notifications;

import lombok.EqualsAndHashCode;
import lombok.Value;

import javax.annotation.Nullable;

/**
 * @author Ruben Gees
 */
@Value
@EqualsAndHashCode(onParam = @__({@Nullable}))
public class NotificationInfo {

    /**
     * Returns the amount of messages.
     */
    private int messageAmount;

    /**
     * Returns the amount of friend requests.
     */
    private int friendRequests;

    /**
     * Returns the amount of news.
     */
    private int news;

    /**
     * Returns the amount of general notifications.
     */
    private int notifications;
}

File: library/src/main/kotlin/me/proxer/library/entity/notifications/NotificationInfo.kt
package me.proxer.library.entity.notifications

import com.squareup.moshi.JsonClass

/**
 * Entity holding counts of notification types.
 *
 * @property messageAmount The amount of messages.
 * @property friendRequestAmount The amount of friend requests.
 * @property newsAmount The amount of news.
 * @property notificationAmount The amount of general notifications.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class NotificationInfo(
    val messageAmount: Int,
    val friendRequestAmount: Int,
    val newsAmount: Int,
    val notificationAmount: Int
)
